nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core-api/src/main/java/org/apache/nifi/controller/AbstractComponentNode.java

```diff
@@ -776,6 +779,23 @@ protected Collection<ValidationResult> computeValidationErrors(final ValidationC
             final Collection<ValidationResult> referencedServiceValidationResults = validateReferencedControllerServices(validationContext);
             validationResults.addAll(referencedServiceValidationResults);
 
+            analyze();
+
+            Optional.ofNullable(getValidationContextFactory().getRuleViolationsManager())
+                .map(ruleViolationsManager -> ruleViolationsManager.getRuleViolationsForSubject(getIdentifier()))
+                .map(Collection::stream)
+                .ifPresent(ruleViolationStream -> ruleViolationStream
+                    .filter(ruleViolation -> ruleViolation.getEnforcementPolicy() == EnforcementPolicy.ENFORCE)
+                    .filter(RuleViolation::isEnabled)
+                    .forEach(ruleViolation -> validationResults.add(
+                        new ValidationResult.Builder()
+                            .subject(getComponent().getClass().getSimpleName())
+                            .valid(false)
+                            .explanation(ruleViolation.getViolationMessage())
+                            .build()
+                    )
+                ));
```

Comment:
It seems like Flow Analysis results should be separate from property validation failures.

Reply:
Again, this makes sense if we consider Validation as the validation of the component itself. Enforcing rules have the duty to invalidate violating components. I see no better place than doing it during validation.

Reply:
I think it does make sense to go ahead and perform the rule analysis here. Since that analysis may be enforced by invalidating the component, we need to ensure that the analysis is performed as part of the validation. If they are performed out of sync, it can lead to problems where a component is valid, started, and then the rule enforcement indicates a problem.